Duplicati/Library/Main/Operation/PurgeFilesHandler.cs

```diff
@@ -123,7 +125,9 @@ private void DoRun(Database.LocalPurgeDatabase db, Library.Utility.IFilter filte
                                 break;
                         }
 
-                        var ts = filesets[ix].Value.AddSeconds(secs);
+                        var tsOriginal = filesets[ix].Value;
+                        var ts = tsOriginal.AddSeconds(secs);

```

Comment:
These changes look ok to me (although I do have questions about the original logic that we have left intact).
1. Does this method assume that `filesets` and `versions` are ordered?  
2. Can the definition of `ts` here result in timestamps in the future?  If so, is that problematic?

Reply:
1. I think it does assume that.  Should I enforce it here too?
2. Not the future. It looks like it is expecting the new fileset to just not be the same timestamp as the original?

Reply:
I think the ts at most would get set to the current time. 

We can look at enforcing sorting within methods but this seems outside the scope of this PR.